### PR TITLE
Refactor MessageAttributes

### PR DESCRIPTION
**Pull request by Admiral-Piett**

### CODE REVIEW

File: app/models/models.go [hunk 1]
```diff
@@ -10,29 +10,28 @@ import (
...
-// TODO - reconcile this with app.SqsMessageAttributeValue - deal with ConvertToOldMessageAttributeValueStructure
-type MessageAttributeValue struct {
-	BinaryListValues []string `json:"BinaryListValues,omitempty"` // currently unsupported by AWS
-	BinaryValue      string   `json:"BinaryValue,omitempty"`
-	DataType         string   `json:"DataType,omitempty"`
-	StringListValues []string `json:"StringListValues,omitempty"` // currently unsupported by AWS
-	StringValue      string   `json:"StringValue,omitempty"`
+type MessageAttribute struct {
+	BinaryListValues []string `json:"BinaryListValues,omitempty" xml:"BinaryListValues,omitempty"` // currently unsupported by AWS
+	BinaryValue      []byte   `json:"BinaryValue,omitempty" xml:"BinaryValue,omitempty"`
```
Comment (Admiral-Piett) [inline app/models/models.go:15]: @kojisaikiAtSony So, this PR is some significant work toward normalizing the MessageAttributes as we previously discussed.  I think this piece specifically will solve your issue in https://github.com/Admiral-Piett/goaws/pull/330.  There's a lot of other overhaul here though I know.

What do you think?  I can build it for you first if you'd like to test it before hand and make sure it works for your use cases.  Let me know!

Comment (kojisaikiAtSony) [reply app/models/models.go:15]: Great! We'll check this work is able to be alternative with #330!

Comment (kojisaikiAtSony) [inline app/models/models.go:15]: The content of this byte array is raw value, not base64 encoded, right?

**Review by kojisaikiAtSony — APPROVED** [review]

**Comment by kojisaikiAtSony** [comment]
@Admiral-Piett It looks good and it works on our app with latest AWS SDK for Java!
You can merge it and release 0.5.1!

**Pull request merged**